.class public Lio/netty/util/concurrent/aa;
.super Ljava/lang/Object;

# interfaces
.implements Lio/netty/util/concurrent/p;


# annotations
.annotation system Ldalvik/annotation/Signature;
    value = {
        "<V:",
        "Ljava/lang/Object;",
        "F::",
        "Lio/netty/util/concurrent/n",
        "<TV;>;>",
        "Ljava/lang/Object;",
        "Lio/netty/util/concurrent/p",
        "<TF;>;"
    }
.end annotation


# static fields
.field private static final a:Lio/netty/util/internal/logging/b;


# instance fields
.field private final b:[Lio/netty/util/concurrent/y;
    .annotation system Ldalvik/annotation/Signature;
        value = {
            "[",
            "Lio/netty/util/concurrent/y",
            "<-TV;>;"
        }
    .end annotation
.end field


# direct methods
.method static constructor <clinit>()V
    .locals 1

    const-class v0, Lio/netty/util/concurrent/aa;

    invoke-static {v0}, Lio/netty/util/internal/logging/c;->a(Ljava/lang/Class;)Lio/netty/util/internal/logging/b;

    move-result-object v0

    sput-object v0, Lio/netty/util/concurrent/aa;->a:Lio/netty/util/internal/logging/b;

    return-void
.end method

.method public varargs constructor <init>([Lio/netty/util/concurrent/y;)V
    .locals 3
    .annotation system Ldalvik/annotation/Signature;
        value = {
            "([",
            "Lio/netty/util/concurrent/y",
            "<-TV;>;)V"
        }
    .end annotation

    .annotation runtime Ljava/lang/SafeVarargs;
    .end annotation

    invoke-direct {p0}, Ljava/lang/Object;-><init>()V

    const-string v0, "promises"

    invoke-static {p1, v0}, Lio/netty/util/internal/k;->a(Ljava/lang/Object;Ljava/lang/String;)Ljava/lang/Object;

    array-length v1, p1

    const/4 v0, 0x0

    :goto_0
    if-ge v0, v1, :cond_1

    aget-object v2, p1, v0

    if-nez v2, :cond_0

    new-instance v0, Ljava/lang/IllegalArgumentException;

    const-string v1, "promises contains null Promise"

    invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V

    throw v0

    :cond_0
    add-int/lit8 v0, v0, 0x1

    goto :goto_0

    :cond_1
    invoke-virtual {p1}, [Lio/netty/util/concurrent/y;->clone()Ljava/lang/Object;

    move-result-object v0

    check-cast v0, [Lio/netty/util/concurrent/y;

    iput-object v0, p0, Lio/netty/util/concurrent/aa;->b:[Lio/netty/util/concurrent/y;

    return-void
.end method


# virtual methods
.method public a(Lio/netty/util/concurrent/n;)V
    .locals 7
    .annotation system Ldalvik/annotation/Signature;
        value = {
            "(TF;)V"
        }
    .end annotation

    const/4 v0, 0x0

    invoke-interface {p1}, Lio/netty/util/concurrent/n;->g_()Z

    move-result v1

    if-eqz v1, :cond_1

    invoke-interface {p1}, Lio/netty/util/concurrent/n;->get()Ljava/lang/Object;

    move-result-object v1

    iget-object v2, p0, Lio/netty/util/concurrent/aa;->b:[Lio/netty/util/concurrent/y;

    array-length v3, v2

    :goto_0
    if-ge v0, v3, :cond_5

    aget-object v4, v2, v0

    invoke-interface {v4, v1}, Lio/netty/util/concurrent/y;->b(Ljava/lang/Object;)Z

    move-result v5

    if-nez v5, :cond_0

    sget-object v5, Lio/netty/util/concurrent/aa;->a:Lio/netty/util/internal/logging/b;

    const-string v6, "Failed to mark a promise as success because it is done already: {}"

    invoke-interface {v5, v6, v4}, Lio/netty/util/internal/logging/b;->c(Ljava/lang/String;Ljava/lang/Object;)V

    :cond_0
    add-int/lit8 v0, v0, 0x1

    goto :goto_0

    :cond_1
    invoke-interface {p1}, Lio/netty/util/concurrent/n;->isCancelled()Z

    move-result v1

    if-eqz v1, :cond_3

    iget-object v2, p0, Lio/netty/util/concurrent/aa;->b:[Lio/netty/util/concurrent/y;

    array-length v3, v2

    move v1, v0

    :goto_1
    if-ge v1, v3, :cond_5

    aget-object v4, v2, v1

    invoke-interface {v4, v0}, Lio/netty/util/concurrent/y;->cancel(Z)Z

    move-result v5

    if-nez v5, :cond_2

    sget-object v5, Lio/netty/util/concurrent/aa;->a:Lio/netty/util/internal/logging/b;

    const-string v6, "Failed to cancel a promise because it is done already: {}"

    invoke-interface {v5, v6, v4}, Lio/netty/util/internal/logging/b;->c(Ljava/lang/String;Ljava/lang/Object;)V

    :cond_2
    add-int/lit8 v1, v1, 0x1

    goto :goto_1

    :cond_3
    invoke-interface {p1}, Lio/netty/util/concurrent/n;->j()Ljava/lang/Throwable;

    move-result-object v1

    iget-object v2, p0, Lio/netty/util/concurrent/aa;->b:[Lio/netty/util/concurrent/y;

    array-length v3, v2

    :goto_2
    if-ge v0, v3, :cond_5

    aget-object v4, v2, v0

    invoke-interface {v4, v1}, Lio/netty/util/concurrent/y;->b(Ljava/lang/Throwable;)Z

    move-result v5

    if-nez v5, :cond_4

    sget-object v5, Lio/netty/util/concurrent/aa;->a:Lio/netty/util/internal/logging/b;

    const-string v6, "Failed to mark a promise as failure because it\'s done already: {}"

    invoke-interface {v5, v6, v4, v1}, Lio/netty/util/internal/logging/b;->d(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/Object;)V

    :cond_4
    add-int/lit8 v0, v0, 0x1

    goto :goto_2

    :cond_5
    return-void
.end method
